File: Euclidean_Algorithm/Euclidean_Algorithm.cs
using System;

namespace Euclid_GCD
{
    class Program
    {
        public static int EuclidGCD(int a, int b)
        {
            if(b == 0)
                return a;
            else
                return EuclidGCD(b, a % b);
        }

       public static void Main (String[] args)
	     {
		       int	a = 315;
		       int	b = 105;
		       int	c = EuclidGCD(a,b);
		       Console.WriteLine(c);
          // prints 105
          // GCD of 315 and 105 is 105
           a = 30;
           b = 105;
           c = EuclidGCD(a,b);
           Console.WriteLine(c);
		      // prints 15
          // GCD of 30 and 105 is 15
           Console.WriteLine();
           Console.ReadLine();
	     }
   }
}

/*
Output:
  105
  15
*/

File: Arithmetic Problems/Euclidean_Algorithm/Euclidean_Algorithm.java

import java.util.*;
import java.lang.*;
import java.io.*;


class Euclid
{
	public static void main (String[] args) throws java.lang.Exception
	{
		int	a = 30;
		int	b = 105;
		int	c = EuclidGCD(a,b);
		System.out.println(c);
		// prints 15
		// GCD of 30 and 105 is 15
	}
	
	public static int EuclidGCD(int a, int b)
	{
		if(b == 0)
			return a;
		else
			return EuclidGCD(b,a%b);
	}
}
